src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx

```diff
@@ -82,61 +166,292 @@ export default function EnvironmentOverview({
     }
 
     const renderAppInfoRow = (item: AppInfoListType, index: number) => {
+        const isSelected = selectedAppIds.includes(item.appId)
         return (
             <div
                 key={`${item.application}-${index}`}
-                className="app-deployments-info-row display-grid dc__align-items-center"
+                className={`app-deployments-info-row dc__w-fit-inherit display-grid dc__align-items-center ${
+                    isHovered === index ? 'bc-n50' : 'bcn-0'
+                } ${lastDeployedClassName}`}
+                onMouseEnter={() => setIsHovered(index)}
+                onMouseLeave={() => setIsHovered(null)}
             >
-                <span className="fs-13 fw-4 cn-7">{item.application}</span>
-                {!isVirtualEnv && (
-                    <AppStatus
-                        appStatus={item.lastDeployed ? item.appStatus : StatusConstants.NOT_DEPLOYED.noSpaceLower}
-                    />
-                )}
+                <div
+                    className={`pl-16 pr-16 app-deployment-info-row-leftsection dc__align-items-center display-grid dc__position-sticky sticky-column ${
+                        isHovered === index ? 'bc-n50' : 'bcn-0'
+                    }`}
+                >
+                    {/* <span> */}
+                    {isHovered !== index && !isSelected ? (
+                        <DevtronIcon className="icon-dim-20" />
+                    ) : (
+                        <label className="dc__position-rel pointer mb-0">
+                            <input
+                                type="checkbox"
+                                className="form__checkbox"
+                                value={item.appId}
+                                onChange={handleSelect}
+                                checked={isSelected}
+                            />
+                            <span className={`form__checkbox-container ${isSelected ? 'tick-icon' : ''}`}></span>
+                        </label>
+                    )}
```

Comment:
Instead of using the isHovered state, we could have used CSS to achieve the same